models/inventory/StockManager.ts

```diff
@@ -126,6 +127,58 @@ export class StockManager {
     throw new ValidationError(t`Both From and To Location cannot be undefined`);
   }
 
+  async #validateBatchWiseStockAvailability(details: SMIDetails) {
```

Comment:
Whatever this function does should be a part of `#validateStockAvailability` there's no need to validate quantity at some location twice.

Reply:
corrected